src/dask_awkward/lib/structure.py

```diff
@@ -924,11 +924,23 @@ def with_name(
 ) -> Array:
     if not highlevel:
         raise ValueError("Only highlevel=True is supported")
+
+    #  TODO: remove once fixed in awkward
+    meta = typetracer_from_form(
+        ak.with_name(
+            array._meta.layout.form.length_zero_array(),
```

Comment:
`array._meta` has a `.behavior` attribute that contains the behavior dict. By unwrapping the meta layout, we never read the behaviors back from this object.

I would assume that we're doing this in other places too. The result of `length_zero_array` needs to be re-wrapped with the original behaviours, perhaps like `ak.Array(array._meta.layout.form.length_zero_array(), behavior=array._meta.behavior)`

Reply:
Has to be tacked on to the result of `typetracer_from_form`, but that effectively did the trick. Have a look!